packages/driver/src/cy/commands/querying.js

```diff
@@ -598,11 +612,57 @@ module.exports = (Commands, Cypress, cy) => {
         cy.once('command:queue:before:end', () => {
           cleanup()
 
-          cy.state('withinSubject', null)
+          state('withinSubject', null)
         })
       }
 
       return subject
     },
   })
+
+  if (Cypress.config('experimentalShadowDomSupport')) {
+    Commands.add('shadow', { prevSubject: 'element' }, (subject, options) => {
+      const userOptions = options || {}
+
+      options = _.defaults({}, userOptions, { log: true })
+
+      const consoleProps = {
+        'Applied To': $dom.getElements(subject),
+      }
+
+      if (options.log !== false) {
+        options._log = Cypress.log({
+          consoleProps () {
+            return consoleProps
+          },
+        })
+      }
+
+      const setEl = ($el) => {
+        if (options.log === false) {
+          return
+        }
+
+        consoleProps.Yielded = $dom.getElements($el)
+        consoleProps.Elements = $el?.length
+
+        return options._log.set({ $el })
+      }
+
+      const getShadowRoots = () => {
+        // find all shadow roots of the subject(s), if any exist
+        const $el = subject
+        .map((i, node) => node.shadowRoot)
```

Comment:
@chrisbreiding how does this work? i had `toArray` here originally to get the native nodes. does `map` call the callback with _both_ the jq wrapped node and the native node?

Reply:
[jQuery's `map`](https://api.jquery.com/map/) calls the callback with the index (`i`) and the native node (`node`). The tricky part I always forget is that the index is the first argument, unlike `map` in lodash, etc.

Reply:
haha they just had to do it differently didn't they :joy: